### PR TITLE
Refactor into separate charts

### PR DESCRIPTION
**Pull request by bheuvel**
Break up the chart into individual charts

Required by bubble toolkit PR https://github.com/MissionCriticalCloud/bubble-toolkit/pull/225
(please merge both at the same time)

### CODE REVIEW

File: charts/cosmic-common/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+apiVersion: v1
+description: Common objects and resources
+name: cosmic-common
```
Comment (eliasgomes) [inline charts/cosmic-common/Chart.yaml:3]: `cosmic-commons` with an `S` ?

File: charts/cosmic-common/templates/_helpers.tpl [hunk 2]
```diff
@@ -0,0 +1,16 @@
+{{/* vim: set filetype=mustache: */}}
```
Comment (eliasgomes) [inline charts/cosmic-common/templates/_helpers.tpl]: These `_helpers.tpl` are not being used, maybe remove them?

File: charts/cosmic-common/templates/configmap.yaml [hunk 3]
```diff
@@ -1,10 +1,12 @@
-{{- define "configmaps.init-container-scripts" -}}
 apiVersion: v1
 kind: ConfigMap
 metadata:
   name: init-container-scripts
+{{- if .Values.namespace }}
```
Comment (eliasgomes) [inline charts/cosmic-common/templates/configmap.yaml]: Just keep the global namespace for simplicity?

File: charts/cosmic-config-server/templates/_cosmic-config-server.json.tpl [hunk 4]
```diff
@@ -0,0 +1,18 @@
+{{- define "secrets.cosmic-config-files.cosmic-config-server.json" -}}
```
Comment (eliasgomes) [inline charts/cosmic-config-server/templates/_cosmic-config-server.json.tpl]: With this new chart structure, I'd suggest to review our template naming strategy.

File: charts/cosmic-config-server/templates/deployment.yaml [hunk 5]
```diff
@@ -1,15 +1,20 @@
-{{- define "deployments.cosmic-config-server" -}}
 apiVersion: extensions/v1beta1
 kind: Deployment
 metadata:
-  name: cosmic-config-server
+  name: {{ .Chart.Name }}
```
Comment (eliasgomes) [inline charts/cosmic-config-server/templates/deployment.yaml]: I'd suggest to remove this helm (deployments and pods) labeling for now.

File: charts/cosmic-config-server/templates/deployment.yaml [hunk 6]
```diff
@@ -34,16 +39,16 @@ spec:
           }]
     spec:
       containers:
-      - name: cosmic-config-server
-        image: {{ .Values.registry }}missioncriticalcloud/cosmic-config-server:{{ .Values.cosmic_config_server.image_tag }}
-        imagePullPolicy: IfNotPresent
+      - name: {{ .Chart.Name }}
+        image: {{ .Values.global.registry }}{{ .Values.image.repository }}:{{ .Values.image.tag }}
+        imagePullPolicy: {{ .Values.image.pullPolicy }}
```
Comment (eliasgomes) [inline charts/cosmic-config-server/templates/deployment.yaml]: We should review our _case_ policy, at the moment we have camel-cased and underscore separated.

**Review by eliasgomes — CHANGES_REQUESTED** [review]

File: charts/cosmic-vault/templates/_helpers.tpl [hunk 7]
```diff
@@ -0,0 +1,16 @@
+{{/* vim: set filetype=mustache: */}}
```
Comment (eliasgomes) [inline charts/cosmic-vault/templates/_helpers.tpl]: There's still a `_helper.tpl` trying to sneak in :)

File: charts/cosmic-vault/templates/deployment.yaml [hunk 8]
```diff
@@ -1,29 +1,39 @@
...
+          value: {{ .Values.global.vault.rootToken }}
+        - name: VAULT_TOKEN
+          value: {{ .Values.global.vault.rootToken }}
+{{- end }}
+{{- end }}
         - name: VAULT_ADDR
           value: http://127.0.0.1:8200
         ports:
-        - containerPort: 8200
+{{- range $index, $node := .Values.service.ports }}
```
Comment (eliasgomes) [inline charts/cosmic-vault/templates/deployment.yaml]: Here we're not using `$index`

File: charts/elasticsearch/templates/deployment.yaml [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+    metadata:
+      labels:
+        app: {{ .Chart.Name }}
+    spec:
+      containers:
+      - name: {{ .Chart.Name }}
+        image: {{ .Values.image.name }}:{{ .Values.image.tag }}
+        imagePullPolicy: {{ .Values.image.pullPolicy }}
+        ports:
+{{- range $index, $node := .Values.service.ports }}
```
Comment (eliasgomes) [inline charts/elasticsearch/templates/deployment.yaml]: Here we're not using `$index`

File: charts/cosmic-usage-api/templates/deployment.yaml [hunk 10]
```diff
@@ -49,9 +48,10 @@ spec:
               name: cosmic-usage-api
               key: config
         ports:
-        - containerPort: 8080
+{{- range $index, $node := .Values.service.ports }}
```
Comment (eliasgomes) [inline charts/cosmic-usage-api/templates/deployment.yaml]: Here we're not using `$index`

File: charts/cosmic-usage-ui/values.yaml [hunk 11]
```diff
@@ -0,0 +1,10 @@
+# Default values for cosmic-usage-ui.
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+replicaCount: 1
+image:
+  name: missioncriticalcloud/cosmic-usage-ui
+  tag: latest
+  pullPolicy: IfNotPresent
+service:
+usageApiBaseUrl:  # for example: https://localhost:8080/   <-- don't forget the slash at the end
```
Comment (eliasgomes) [inline charts/cosmic-usage-ui/values.yaml:10]: New line missing

File: charts/cosmic-usage-ui/templates/deployment.yaml [hunk 12]
```diff
@@ -31,16 +30,15 @@ spec:
...
         env:
         - name: COSMIC_USAGE_API_BASE_URL
-          value: {{ .Values.cosmic_usage_ui.usage_api_base_url }}
+          value: {{ .Values.usageApiBaseUrl }}
       volumes:
       - name: init-container-scripts
         configMap:
-          name: init-container-scripts
-{{- end -}}
+          name: init-container-scripts
```
Comment (eliasgomes) [inline charts/cosmic-usage-ui/templates/deployment.yaml]: New line missing

File: charts/cosmic-vault/templates/deployment.yaml [hunk 13]
```diff
@@ -46,6 +56,4 @@ spec:
       volumes:
       - name: cosmic-vault-files
         configMap:
-          name: cosmic-vault-files
-{{- end -}}
-{{- end -}}
+          name: cosmic-vault-files
```
Comment (eliasgomes) [inline charts/cosmic-vault/templates/deployment.yaml]: New line missing

File: charts/elasticsearch/values.yaml [hunk 14]
```diff
@@ -0,0 +1,21 @@
...
+  ports:
+    - name: http
+      port: 9200
+      protocol: TCP
+      nodePort: 31102
+    - name: transport
+      port: 9300
+      protocol: TCP
+
+
```
Comment (eliasgomes) [inline charts/elasticsearch/values.yaml]: Extra new line :D

File: charts/logstash/templates/deployment.yaml [hunk 15]
```diff
@@ -1,30 +1,29 @@
...
+        app: {{ template "fullname" . }}
     spec:
       containers:
-      - name: logstash
-        image: logstash:{{ .Values.logstash.image_tag }}
-{{- if .Values.dev_mode }}
+      - name: {{ .Chart.Name }}
+        image: "{{ .Values.image.name }}:{{ .Values.image.tag }}"
+        imagePullPolicy: {{ .Values.image.pullPolicy }}
+{{- if or .Values.devMode (and .Values.global .Values.global.devMode) }}
```
Comment (eliasgomes) [inline charts/logstash/templates/deployment.yaml]: Same places we're checking for `devMode` like `if .Values.global.devMode` and some places like this: `if or .Values.devMode (and .Values.global .Values.global.devMode)`. Maybe using only the global one?

File: charts/cosmic-metrics-collector/values.yaml [hunk 16]
```diff
@@ -0,0 +1,11 @@
...
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+replicaCount: 1
+image:
+  name: missioncriticalcloud/cosmic-metrics-collector
+  tag: latest
+  pullPolicy: IfNotPresent
+service:
+
+
```
Comment (eliasgomes) [inline charts/cosmic-metrics-collector/values.yaml]: Extra new line

**Review by eliasgomes — CHANGES_REQUESTED** [review]

File: charts/cosmic-usage-ui/values.yaml (hunk 11 above)
Comment (bheuvel) [reply charts/cosmic-usage-ui/values.yaml:10]: Added...

**Comment by bheuvel** [comment]
@eliasgomes a.k.a. "hawk eye" 😉 , thanks for the review. Adjusted additional comments.

**Review by eliasgomes — APPROVED** [review]

**Comment by eliasgomes** [comment]
@bheuvel thanks for patience! :P

**Pull request merged**